protocol-jackson/src/main/java/io/camunda/zeebe/protocol/jackson/record/LazyJsonSerializable.java

```diff
@@ -0,0 +1,25 @@
+package io.camunda.zeebe.protocol.jackson.record;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.camunda.zeebe.protocol.record.JsonSerializable;
+import java.io.UncheckedIOException;
+import org.immutables.value.Value;
+
+public interface LazyJsonSerializable extends JsonSerializable {
```

Comment:
I opted against the abstract class and went with an interface. This is necessary to ensure the `toJson` property is auxilliary - that is, Immutables does not generate any `toJson` property, field, accessors, etc. So we mark it as `lazy`, which means it's computed once when called, but memoized. We could also mark it as `derived`, so it would be computed all the time, that's also fine. I pre-emptively mark them as deprecated as for this implementation, people should just use a configured `ObjectMapper` instead of this.

Reply:
:question: Also, I debated making it public - technically it's exposed anyway since all the types implement it, so...meh. But it might be more explicit to not make it public if only so people aren't tempted to implement it :smile:

Reply:
This was removed so... :man_shrugging:

Reply:
❓ This is no longer relevant right?